subsys/bluetooth/audio/broadcast_sink.c

```diff
@@ -463,6 +463,14 @@ static bool pa_decode_base(struct bt_data *data, void *user_data)
 
 	codec_qos.pd = net_buf_simple_pull_le24(&net_buf);
 	sink->subgroup_count = net_buf_simple_pull_u8(&net_buf);
+
```

Comment:
Yes, when looking at this for this PR, I realized that we could probably just sets the array sizes to 0 and then simply cast the adv report to a BASE. Will give this a try tomorrow :)

Reply:
Hmm, the `struct bt_audio_base` ends up relying on the `bt_codec_data` which has a fixed size, so I can't trivially modify it to not be a specific size. 

I think it would make sense to investigate this optimization/improvement in a new issue, as it would require a significant change in the BAP implementation across all roles.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/44113

Reply:
There is a limit of 32 BISes, isn't there?  And there shall be at least one BIS per subgroup.  Which means that the upper limit on the number of subgroups is also 32.
But the size of the subgroup itself is not given.

Reply:
We check `sink->subgroup_count > ARRAY_SIZE(base.subgroups)` right below. Currently we only support a single subgroup, so that's a pretty strict check. Once we support more, that will also verify the subgroup count, unless we implement https://github.com/zephyrproject-rtos/zephyr/issues/44113 first, and in that case we probably need to very the subgroup count value.